tests/foreman/cli/test_docker.py

```diff
@@ -37,50 +37,101 @@
 from robottelo.datafactory import valid_docker_upstream_names
 from robottelo.decorators import skip_if_not_set
-from robottelo.test import CLITestCase
-from robottelo.vm import VirtualMachine
 
-DOCKER_PROVIDER = 'Docker'
-REPO_CONTENT_TYPE = 'docker'
-REPO_UPSTREAM_NAME = 'busybox'
+BASE_URL = 'http://download.eng.rdu2.redhat.com/rhel-7/rel-eng'
+SERVER_REPO = f'{BASE_URL}/updates/RHEL-7/latest-RHEL-7.9/compose/Server/x86_64/os/'
+OPTIONAL_REPO = f'{BASE_URL}/RHEL-7/latest-RHEL-7/compose/Server-optional/x86_64/os/'
+EXTRAS_REPO = f'{BASE_URL}/EXTRAS-7/latest-EXTRAS-7.9-RHEL-7/compose/Server/x86_64/os/'
 
+logger = logging.getLogger('robottelo')
 
-def _make_docker_repo(product_id, name=None, upstream_name=None, url=None):
+
+def _repo(product_id, name=None, upstream_name=None, url=None):
     """Creates a Docker-based repository.
 
     :param product_id: ID of the ``Product``.
     :param str name: Name for the repository. If ``None`` then a random
         value will be generated.
     :param str upstream_name: A valid name of an existing upstream repository.
-        If ``None`` then defaults to ``busybox``.
+        If ``None`` then defaults to DOCKER_UPSTREAM_NAME constant.
     :param str url: URL of repository. If ``None`` then defaults to
         DOCKER_REGISTRY_HUB constant.
     :return: A ``Repository`` object.
     """
     return make_repository(
         {
-            'content-type': REPO_CONTENT_TYPE,
-            'docker-upstream-name': upstream_name or REPO_UPSTREAM_NAME,
-            'name': name or choice(generate_strings_list(15, ['numeric', 'html'])),
+            'content-type': REPO_TYPE['docker'],
+            'docker-upstream-name': upstream_name or DOCKER_UPSTREAM_NAME,
+            'name': name or gen_string('alpha', 5),
             'product-id': product_id,
             'url': url or DOCKER_REGISTRY_HUB,
         }
     )
 
 
-class DockerManifestTestCase(CLITestCase):
+def _content_view(repo_id, org_id):
+    """Create a content view and link it to the given repository."""
+    content_view = make_content_view({'composite': False, 'organization-id': org_id})
+    ContentView.add_repository({'id': content_view['id'], 'repository-id': repo_id})
+    return ContentView.info({'id': content_view['id']})
+
+
+@pytest.fixture
+def docker_host(rhel7_contenthost):
+    """Instantiate and set up a docker host VM. Destroy VM when done."""
+    logger.info('Installing katello-ca on the external docker host')
+    rhel7_contenthost.install_katello_ca()
+
+    repos = {'server': SERVER_REPO, 'optional': OPTIONAL_REPO, 'extras': EXTRAS_REPO}
+    rhel7_contenthost.create_custom_repos(**repos)
+    rhel7_contenthost.execute('yum -y install docker')
+    rhel7_contenthost.execute('systemctl start docker')
+    return rhel7_contenthost
+
+
+@pytest.fixture
+def repo(module_product):
+    return _repo(module_product.id)
+
+
+@pytest.fixture
+def content_view(module_org, repo):
+    return _content_view(repo['id'], module_org.id)
+
```

Comment:
Use global API based fixtures if not create them there and use it here.

Reply:
Similar setup fixtures you have created in API tests. Creating a global API fixtures can be used in CLI, API and UI tests.